WordPress/src/test/java/org/wordpress/android/ui/reader/usecases/ReaderFetchRelatedPostsUseCaseTest.kt

```diff
@@ -0,0 +1,77 @@
+package org.wordpress.android.ui.reader.usecases
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.whenever
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.InternalCoroutinesApi
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Before
+
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mock
+import org.mockito.junit.MockitoJUnitRunner
+import org.wordpress.android.MainCoroutineScopeRule
+import org.wordpress.android.models.ReaderPost
+import org.wordpress.android.test
+import org.wordpress.android.ui.reader.ReaderEvents.RelatedPostsUpdated
+import org.wordpress.android.ui.reader.actions.ReaderPostActionsWrapper
+import org.wordpress.android.ui.reader.usecases.ReaderFetchRelatedPostsUseCase.FetchRelatedPostsState
+import org.wordpress.android.ui.reader.usecases.ReaderFetchRelatedPostsUseCase.FetchRelatedPostsState.Failed
+import org.wordpress.android.util.NetworkUtilsWrapper
+
+@InternalCoroutinesApi
+@RunWith(MockitoJUnitRunner::class)
+class ReaderFetchRelatedPostsUseCaseTest {
+    @Rule
+    @JvmField val rule = InstantTaskExecutorRule()
+
+    @ExperimentalCoroutinesApi
+    @Rule
+    @JvmField val coroutineScope = MainCoroutineScopeRule()
+
+    lateinit var useCase: ReaderFetchRelatedPostsUseCase
+    @Mock lateinit var readerPostActionsWrapper: ReaderPostActionsWrapper
+    @Mock lateinit var networkUtilsWrapper: NetworkUtilsWrapper
+    @Mock lateinit var readerPost: ReaderPost
+
+    @Before
+    fun setup() {
+        useCase = ReaderFetchRelatedPostsUseCase(networkUtilsWrapper, readerPostActionsWrapper)
+
+        whenever(networkUtilsWrapper.isNetworkAvailable()).thenReturn(true)
+    }
+
```

Comment:
Minor (🔍): Is it possible for us to also test the `AlreadyRunning` logic somehow?

Reply:
I tried to include a test but couldn't do it with event bus.  

We can test it manually by commenting below line [here](https://github.com/wordpress-mobile/WordPress-Android/pull/14142/files/087f3d6cee3ba37b0ff4ec1755a1ee5a59aee046#diff-7b33e01a0849528abbcbf8188cef615b72a21a6e2cf3656101b7bc8e0a454496R58):
` continuations[request] = null`

and fetching related posts twice from `ReaderPostDetailViewModel.onRelatedPostsRequested`.

Reply:
Cool, great, yes, having fun with event bus! 😞